gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreHeadSnapshot.java

```diff
@@ -0,0 +1,17 @@
+package com.virtuslab.gitcore.api;
+
+import io.vavr.collection.List;
+import io.vavr.control.Option;
+
+/**
+ * An immutable snapshot of git repository's HEAD for some specific moment in time.
+ * Each {@code get...} method is guaranteed to return the same value each time it's called on a given object.
+ */
+public interface IGitCoreHeadSnapshot {
```

Comment:
![image](https://user-images.githubusercontent.com/3383210/86473712-baa32780-bd41-11ea-82a2-cbca0b077f2e.png)

Reply:
This red logo text and the strike outgoing from bottom left to the upper right reminds me of something :rofl: :flight_departure:

Reply:
This red logo text and the strike outgoing from bottom left to the upper right reminds me of something :rofl: :flight_departure:

Reply:
Wow it's actually an **Indonesian** movie

![image](https://user-images.githubusercontent.com/3383210/86630520-395fc500-bfcd-11ea-85cd-682472be6394.png)

Reply:
And fun fact, high in IMDb results there's also 2011 Thai _Headshot_ xD

![image](https://user-images.githubusercontent.com/3383210/86630963-d15dae80-bfcd-11ea-94d2-00f100f20db4.png)